backend/router/validator.py

```
@@ -906,43 +1083,476 @@ def _execute_move(
+        target_entities = [state.entities[t] for t in targets]
+
+        actor_entity = state.entities[actor]
+        primary_target_entity = target_entities[0]
+
+        # Validate actor is a creature that can talk
+        if actor_entity.type not in ("pc", "npc"):
+            return ToolResult(
+                ok=False,
+                tool_id="ask_clarifying",
+                args={
+                    "question": f"Only characters can talk to others.",
```

Ruff (0.13.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)